### ISSUE TITLE
`Literal[0]` cannot be assigned to declared type `InitVar[int]`

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#3095]
```python
from dataclasses import dataclass, InitVar

@dataclass
class C:
    database: InitVar[int] = 0
```

Using the latest & greatest pyright 1.1.223. Looking at #174 this appears to be a regression.

(mypy doesn't complain but that's apparently because it just treats `InitVar` as `Any`)

**Comment by erictraut** [comment]
I'm not able to repro any error with the above code. Could you confirm that you're seeing an error?

**Comment by not-my-profile** [comment]
Yes I get an error:

```
$ pyright foo.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/foo.py
  /tmp/foo.py:5:30 - error: Expression of type "Literal[0]" cannot be assigned to declared type "InitVar[int]"
    "Literal[0]" is incompatible with "InitVar[int]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.654sec
```

**Comment by erictraut** [comment]
I still can't repro it. I suspect you have a package called `dataclasses` installed in your Python environment that is overriding the stdlib library. Please check for that. 

If that doesn't explain it, please try to repro the problem with a clean virtual environment. I am quite certain this isn't a bug in pyright but rather something in your environment that's resulting in this error condition.

**Comment by not-my-profile** [comment]
Ah, yes thank you! I indeed had `types-dataclasses` installed which was causing this error.
